luthier/src/main/java/com/yahoo/luthier/webservice/data/config/metric/MetricMakerDictionary.java

```diff
@@ -0,0 +1,213 @@
+     * @param metricMaker value to add to dictionary
+     * @return <tt>true</tt> if the dictionary did not already contain the specified metric maker
+     */
+    public boolean add(String name, MetricMaker metricMaker) {
+        if (nameToMetricMaker.containsKey(name.toLowerCase(Locale.ENGLISH))) {
+            return false;
+        }
+        MetricMaker metricMakers = nameToMetricMaker.put(name.toLowerCase(Locale.ENGLISH), metricMaker);
+        if (metricMakers != null) {
+            // should never happen unless multiple loaders are running in race-condition
+            ConcurrentModificationException e = new ConcurrentModificationException();
+            LOG.error("Multiple loaders updating MetricMakerDictionary", e);
+            throw e;
+        }
+        return true;
+    }
+
+    /**
+     * Adds all of the metric makers in the specified collection to the dictionary.
+     *
+     * @param metricMakers collection of metric makers to add
+     * @return <tt>true</tt> if the dictionary changed as a result of the call
+     */
+    public boolean addAll(Collection<MetricMaker> metricMakers) {
+        boolean flag = false;
+        for (MetricMaker metricMaker : metricMakers) {
+            flag = add(metricMaker.toString(), metricMaker) || flag;
+        }
+        return flag;
+    }
+
+    @Override
+    public String toString() {
+        return "MetricMaker Dictionary: " + nameToMetricMaker;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(nameToMetricMaker);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj instanceof MetricMakerDictionary) {
+            MetricMakerDictionary that = (MetricMakerDictionary) obj;
+            return nameToMetricMaker.equals(that.nameToMetricMaker);
+        }
+        return false;
+    }
+
+    /**
+     * Parse parameters for maker's constructor based on parameter's name and type.
+     *
+     * @param paramType type of the parameter in maker's constructor
+     * @param paramName name of the parameter in maker's constructor
+     * @param maker     the maker template used to find parameter's value by name
+     * @return the value of parameter (can be any type)
+     */
+    private Object parseParams(String paramType, String paramName, MetricMakerTemplate maker) {
+        if ("MetricDictionary".equals(paramType)) {
+            return metricDictionary;
+        }
+        if ("DimensionDictionary".equals(paramType)) {
+            return dimensionDictionary;
+        }
+        if ("ZonelessTimeGrain".equals(paramType)) {
+            return DefaultTimeGrain.valueOf(maker.getParams().get(paramName));
+        }
+        if ("ArithmeticPostAggregationFunction".equals(paramType)) {
+            return ArithmeticPostAggregation
+                    .ArithmeticPostAggregationFunction
+                    .valueOf(maker.getParams().get(paramName));
```

Comment:
for enum types you should not be doing a string matching. The selected candidate constructor should already have the enum type that you matched to. You should be able to use the enum type from the candidate constructor along with the param value from the config to get the actual enum value to pass to your candidate constructor.